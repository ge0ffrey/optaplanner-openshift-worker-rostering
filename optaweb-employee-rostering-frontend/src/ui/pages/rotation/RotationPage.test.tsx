/*
 * Copyright 2019 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

describe('Rotation Page', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('TODO: REPLACE ME WITH REAL TESTS', () => {
    /* REPLACE ME PLEASE */
  });
});

export {};
/*
const spot: Spot = {
  tenantId: 0,
  id: 2,
  version: 0,
  name: 'Spot',
  requiredSkillSet: [
    {
      tenantId: 0,
      id: 3,
      version: 0,
      name: 'Skill',
    },
  ],
};

const newSpot: Spot = {
  ...spot,
  id: 111,
  name: 'New Spot',
};

const employee: Employee = {
  tenantId: 0,
  id: 4,
  version: 0,
  name: 'Employee 1',
  contract: {
    tenantId: 0,
    id: 5,
    version: 0,
    name: 'Basic Contract',
    maximumMinutesPerDay: 10,
    maximumMinutesPerWeek: 70,
    maximumMinutesPerMonth: 500,
    maximumMinutesPerYear: 6000,
  },
  skillProficiencySet: [{
    tenantId: 0,
    id: 6,
    version: 0,
    name: 'Not Required Skill',
  }],
  shortId: 'e1',
  color: '#FFFFFF',
};

const shiftTemplate: ShiftTemplate = {
  tenantId: 0,
  id: 1,
  version: 0,
  durationBetweenRotationStartAndTemplateStart: moment.duration(1, 'day').add(9, 'hours'),
  shiftTemplateDuration: moment.duration(8, 'hours'),
  spot,
  requiredSkillSet: [],
  rotationEmployee: employee,
};

const rosterState: RosterState = {
  tenant: {
    id: 0,
    version: 0,
    name: 'Tenant',
  },
  publishNotice: 14,
  publishLength: 7,
  firstDraftDate: new Date('2018-07-01'),
  draftLength: 7,
  unplannedRotationOffset: 0,
  rotationLength: 7,
  lastHistoricDate: new Date('2018-07-01'),
  timeZone: 'EST',
}; */
