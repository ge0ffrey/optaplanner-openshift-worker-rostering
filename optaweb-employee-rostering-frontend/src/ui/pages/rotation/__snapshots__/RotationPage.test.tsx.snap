// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rotation Page should render correctly when creating a new shift template via button 1`] = `
<Fragment>
  <Level
    gutter="sm"
    style={
      Object {
        "backgroundColor": "var(--pf-global--BackgroundColor--100)",
        "height": "60px",
        "padding": "5px 5px 5px 5px",
      }
    }
  >
    <LevelItem
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <TypeaheadSelectInput
        aria-label="Select Spot"
        emptyText="Trans(i18nKey=selectSpot)"
        noClearButton={true}
        onChange={[Function]}
        optionToStringMap={[Function]}
        options={
          Array [
            Object {
              "covidWard": false,
              "id": 2,
              "name": "Spot",
              "requiredSkillSet": Array [
                Object {
                  "id": 3,
                  "name": "Skill",
                  "tenantId": 0,
                  "version": 0,
                },
              ],
              "tenantId": 0,
              "version": 0,
            },
            Object {
              "covidWard": false,
              "id": 111,
              "name": "New Spot",
              "requiredSkillSet": Array [
                Object {
                  "id": 3,
                  "name": "Skill",
                  "tenantId": 0,
                  "version": 0,
                },
              ],
              "tenantId": 0,
              "version": 0,
            },
          ]
        }
        value={
          Object {
            "covidWard": false,
            "id": 2,
            "name": "Spot",
            "requiredSkillSet": Array [
              Object {
                "id": 3,
                "name": "Skill",
                "tenantId": 0,
                "version": 0,
              },
            ],
            "tenantId": 0,
            "version": 0,
          }
        }
      />
      <Component
        itemCount={7}
        onSetPage={[Function]}
        page={1}
        perPage={7}
        perPageOptions={Array []}
        titles={
          Object {
            "currPage": "Trans(i18nKey=currentWeek)",
            "items": "Trans(i18nKey=days)",
            "itemsPerPage": "Trans(i18nKey=weekNum)",
            "optionsToggle": "Trans(i18nKey=select)",
            "page": "Trans(i18nKey=week)",
            "paginationTitle": "Trans(i18nKey=weekSelect)",
            "perPageSuffix": "Trans(i18nKey=weekNum)",
            "toFirstPage": "Trans(i18nKey=gotoFirstWeek)",
            "toLastPage": "Trans(i18nKey=gotoLastWeek)",
            "toNextPage": "Trans(i18nKey=gotoNextWeek)",
            "toPreviousPage": "Trans(i18nKey=gotoPreviousWeek)",
          }
        }
      />
    </LevelItem>
    <LevelItem
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <Component
        aria-label="Create Shift Template"
        onClick={[Function]}
        style={
          Object {
            "margin": "5px",
          }
        }
      >
        Trans(i18nKey=createShiftTemplate)
      </Component>
    </LevelItem>
  </Level>
  <Component
    aria-label="Edit Shift Template"
    isOpen={true}
    onClose={[Function]}
    onDelete={[Function]}
    onSave={[Function]}
  />
  <Schedule
    dateFormat={[Function]}
    dayStyle={[Function]}
    endAccessor={[Function]}
    endDate={2018-01-07T00:00:00.000Z}
    eventComponent={[Function]}
    eventStyle={[Function]}
    events={
      Array [
        Object {
          "end": 2018-01-01T17:00:00.000Z,
          "shiftTemplate": Object {
            "durationBetweenRotationStartAndTemplateStart": "P1DT9H",
            "id": 1,
            "requiredSkillSet": Array [],
            "rotationEmployee": Object {
              "contract": Object {
                "id": 5,
                "maximumMinutesPerDay": 10,
                "maximumMinutesPerMonth": 500,
                "maximumMinutesPerWeek": 70,
                "maximumMinutesPerYear": 6000,
                "name": "Basic Contract",
                "tenantId": 0,
                "version": 0,
              },
              "covidRiskType": "INOCULATED",
              "id": 4,
              "name": "Employee 1",
              "skillProficiencySet": Array [
                Object {
                  "id": 6,
                  "name": "Not Required Skill",
                  "tenantId": 0,
                  "version": 0,
                },
              ],
              "tenantId": 0,
              "version": 0,
            },
            "shiftTemplateDuration": "PT8H",
            "spot": Object {
              "covidWard": false,
              "id": 2,
              "name": "Spot",
              "requiredSkillSet": Array [
                Object {
                  "id": 3,
                  "name": "Skill",
                  "tenantId": 0,
                  "version": 0,
                },
              ],
              "tenantId": 0,
              "version": 0,
            },
            "tenantId": 0,
            "version": 0,
          },
          "start": 2018-01-01T09:00:00.000Z,
        },
      ]
    }
    key="2"
    onAddEvent={[Function]}
    onUpdateEvent={[Function]}
    popoverBody={[Function]}
    popoverHeader={[Function]}
    startAccessor={[Function]}
    startDate={2017-12-31T00:00:00.000Z}
    titleAccessor={[Function]}
    wrapperStyle={[Function]}
  />
</Fragment>
`;

exports[`Rotation Page should render correctly when loaded 1`] = `
<Fragment>
  <Level
    gutter="sm"
    style={
      Object {
        "backgroundColor": "var(--pf-global--BackgroundColor--100)",
        "height": "60px",
        "padding": "5px 5px 5px 5px",
      }
    }
  >
    <LevelItem
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <TypeaheadSelectInput
        aria-label="Select Spot"
        emptyText="Trans(i18nKey=selectSpot)"
        noClearButton={true}
        onChange={[Function]}
        optionToStringMap={[Function]}
        options={
          Array [
            Object {
              "covidWard": false,
              "id": 2,
              "name": "Spot",
              "requiredSkillSet": Array [
                Object {
                  "id": 3,
                  "name": "Skill",
                  "tenantId": 0,
                  "version": 0,
                },
              ],
              "tenantId": 0,
              "version": 0,
            },
            Object {
              "covidWard": false,
              "id": 111,
              "name": "New Spot",
              "requiredSkillSet": Array [
                Object {
                  "id": 3,
                  "name": "Skill",
                  "tenantId": 0,
                  "version": 0,
                },
              ],
              "tenantId": 0,
              "version": 0,
            },
          ]
        }
        value={
          Object {
            "covidWard": false,
            "id": 2,
            "name": "Spot",
            "requiredSkillSet": Array [
              Object {
                "id": 3,
                "name": "Skill",
                "tenantId": 0,
                "version": 0,
              },
            ],
            "tenantId": 0,
            "version": 0,
          }
        }
      />
      <Component
        itemCount={7}
        onSetPage={[Function]}
        page={1}
        perPage={7}
        perPageOptions={Array []}
        titles={
          Object {
            "currPage": "Trans(i18nKey=currentWeek)",
            "items": "Trans(i18nKey=days)",
            "itemsPerPage": "Trans(i18nKey=weekNum)",
            "optionsToggle": "Trans(i18nKey=select)",
            "page": "Trans(i18nKey=week)",
            "paginationTitle": "Trans(i18nKey=weekSelect)",
            "perPageSuffix": "Trans(i18nKey=weekNum)",
            "toFirstPage": "Trans(i18nKey=gotoFirstWeek)",
            "toLastPage": "Trans(i18nKey=gotoLastWeek)",
            "toNextPage": "Trans(i18nKey=gotoNextWeek)",
            "toPreviousPage": "Trans(i18nKey=gotoPreviousWeek)",
          }
        }
      />
    </LevelItem>
    <LevelItem
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <Component
        aria-label="Create Shift Template"
        onClick={[Function]}
        style={
          Object {
            "margin": "5px",
          }
        }
      >
        Trans(i18nKey=createShiftTemplate)
      </Component>
    </LevelItem>
  </Level>
  <Component
    aria-label="Edit Shift Template"
    isOpen={false}
    onClose={[Function]}
    onDelete={[Function]}
    onSave={[Function]}
  />
  <Schedule
    dateFormat={[Function]}
    dayStyle={[Function]}
    endAccessor={[Function]}
    endDate={2018-01-07T00:00:00.000Z}
    eventComponent={[Function]}
    eventStyle={[Function]}
    events={
      Array [
        Object {
          "end": 2018-01-01T17:00:00.000Z,
          "shiftTemplate": Object {
            "durationBetweenRotationStartAndTemplateStart": "P1DT9H",
            "id": 1,
            "requiredSkillSet": Array [],
            "rotationEmployee": Object {
              "contract": Object {
                "id": 5,
                "maximumMinutesPerDay": 10,
                "maximumMinutesPerMonth": 500,
                "maximumMinutesPerWeek": 70,
                "maximumMinutesPerYear": 6000,
                "name": "Basic Contract",
                "tenantId": 0,
                "version": 0,
              },
              "covidRiskType": "INOCULATED",
              "id": 4,
              "name": "Employee 1",
              "skillProficiencySet": Array [
                Object {
                  "id": 6,
                  "name": "Not Required Skill",
                  "tenantId": 0,
                  "version": 0,
                },
              ],
              "tenantId": 0,
              "version": 0,
            },
            "shiftTemplateDuration": "PT8H",
            "spot": Object {
              "covidWard": false,
              "id": 2,
              "name": "Spot",
              "requiredSkillSet": Array [
                Object {
                  "id": 3,
                  "name": "Skill",
                  "tenantId": 0,
                  "version": 0,
                },
              ],
              "tenantId": 0,
              "version": 0,
            },
            "tenantId": 0,
            "version": 0,
          },
          "start": 2018-01-01T09:00:00.000Z,
        },
      ]
    }
    key="2"
    onAddEvent={[Function]}
    onUpdateEvent={[Function]}
    popoverBody={[Function]}
    popoverHeader={[Function]}
    startAccessor={[Function]}
    startDate={2017-12-31T00:00:00.000Z}
    titleAccessor={[Function]}
    wrapperStyle={[Function]}
  />
</Fragment>
`;

exports[`Rotation Page should render correctly when loading 1`] = `
<EmptyState
  variant="full"
>
  <EmptyStateIcon
    icon={[Function]}
  />
  <Trans
    components={
      Array [
        <Title
          headingLevel="h5"
          size="lg"
        />,
        <EmptyStateBody />,
        <Unknown
          aria-label="Spots Page"
          onClick={[Function]}
          variant="primary"
        />,
      ]
    }
    i18nKey="noSpots"
    t={[MockFunction]}
  />
</EmptyState>
`;
